### ISSUE TITLE
Notifications for image downloads are inconsistent

### ISSUE THREAD

**Issue by touzainanboku** [ccrama/Slide#1966]
Slide version: 5.4
Android version: 6.0.1
- According to #1959, the app is supposed to show a separate notification for each image both while downloading and after.
- Based on ccrama's comment on #1894, the app was supposed to show a notification immediately after the download button is pressed. 

I'm having inconsistency in both of those, as seen in the video [here](https://streamable.com/0pge). 

- When the first image is downloaded, the notification appears only after the image has finished downloading. This doesn't always happen, but as seen here, it sometimes does. Even if it does appear early enough to show download progress, it's still not instantaneous in creating the notification. 
- For all the images after, no separate notification is created, and again, no download progress is shown. Unless, this is intended behavior and separate notifications aren't supposed to be made for images in a single album, in which case I'd like to request each image in an album to be separated as well. Although, I'm still facing issues with this even when images aren't from a single album. Sometimes they create a separate notification, sometimes they override the previous one. 

Also, is there supposed to be a timeout for notifications for finished downloads? Occasionally they quickly disappear within seconds. I actually just downloaded an image and stared at the screen for a few seconds, and the notification dismissed itself. Other times, they stay there until I dismiss them.

**Comment by ccrama** [comment]
I have never seen this behavior before and can't reproduce it. Are you sure you have no xposed modules or other mods that would affect notification display?

**Comment by touzainanboku** [comment]
I checked my modules just in case but I didn't find anything that seems to be interfering with notifications. Still, I'll see if I can reproduce it on a stock device.

**Comment by touzainanboku** [comment]
OK, so I just updated my phone so I tried this again before I modify anything, and I was able to reproduce it again. Exact same behavior. Notification doesn't appear until the download is finished, and the downloads after that don't create separate notifications. 
Again, doesn't happen all the time, only sometimes. Still, it's just a small nuisance, and it doesn't detract anything from Slide's awesomeness.

**Comment by ccrama** [comment]
So turns out Services are natural singletons (wasn't aware), so every time we started the image saving service it would overwrite the old variables. Has been fixed!

**Comment by touzainanboku** [comment]
Does this mean the new 5.4.2-alpha3 shouldn't have this bug? Cause I'm still experiencing it.

**Comment by ccrama** [comment]
Sometimes it is loading the image from cache and not downloading it, in that case it might not have time to show the progress. And the album downloads are separate from the way image downloads work, you can make a new issue for that if you'd like

**Comment by touzainanboku** [comment]
> Sometimes it is loading the image from cache and not downloading it, in that case it might not have time to show the progress.

That would make sense (I'd still like to see a notification for it immediately though, even if it is gonna take a little bit for it to actually save, but that's a different matter entirely). However, in the cases I've noticed, I actually only had the LQ version cached, and I actually did see an increase in network activity, so that can't be it. You can actually see this in the video I posted, although it's only for a second, but every time I tap download the internet speed spikes up a little and then goes back to zero. 
Also, it's not that I want to see the progress per se, it's just that I find not getting any clear indication that the app has received my input frustrating, especially combining with #1967, which you said isn't fixable. Basically, it's that:
- even if the notification does correctly appear, it's not as immediate as I'd like, meaning I have to wait a little to be sure it started
- sometimes it only appears after it finished downloading, which makes the wait time even longer, and it's not because it's loading from cache either


> you can make a new issue for that if you'd like

OK. I'll make a request for separate notifications then. Thanks for the clarification. 

I'm sorry I keep making new issues about the same thing, but it's just that the app is well polished in general, especially with how smooth it is with the new alpha, that image downloading really sticks out to me. I just personally don't find its behavior consistent and predictable enough. Thanks again for replying.

**Comment by ccrama** [comment]
I really cannot reproduce the issue you're describing at the top though, right as I click the button, even on LQ mode, it shows the download progress and afterward shows the correct notif. I'm thinking there has to be something else interfering with notifications on your device as the behavior you're describing can no longer be reproduced. I was able to reproduce it before alpha 3 but no longer
